city/routers.py

```diff
@@ -0,0 +1,61 @@
+from dependencies import get_db
+from fastapi import APIRouter, Depends, Response, status, HTTPException
+from settings import settings
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from city import schemas, crud
+
+router = APIRouter(prefix=f"{settings.API_PREFIX}/cities")
+
+
+@router.get("/", response_model=list[schemas.City])
+async def read_cities(db: AsyncSession = Depends(get_db)):
+    return await crud.get_all_cities(db=db)
+
+
+@router.post(
+    "/",
+    response_model=schemas.City,
+    status_code=status.HTTP_201_CREATED
+)
+async def new_city(
+        city: schemas.CityCreate,
+        db: AsyncSession = Depends(get_db)
+):
+    return await crud.create_city(db=db, city=city)
+
+
+@router.get("/{city_id}/", response_model=schemas.City)
+async def read_city(city_id: int, db: AsyncSession = Depends(get_db)):
+    city = await crud.retrieve_city(db=db, id=city_id)
+    if city is None:
+        raise HTTPException(status_code=404, detail="City not found")
+    return city
+
+
+@router.put("/{city_id}/", response_model=schemas.City)
+async def update_city(
+        city_id: int,
+        city: schemas.CityCreate,
```

Comment:
The `update_city` function uses `schemas.CityCreate` as the input model. Ensure that this model is appropriate for updates. If partial updates are allowed, consider using a different schema that allows optional fields or a `schemas.CityUpdate` model.